### PR TITLE
Tests: autoformat with prettier

### PR DESCRIPTION
**Pull request by samouri**
Autoformat all tests files with `calypso-prettier`

I generated this PR by running a custom prettier with the combine pragmas modification.
I wouldn't want to merge this before calypso-prettier is updated in `wp-calypso`, but this PR serves as a good test for both

```
../calypso-prettier/bin/prettier.js  --write server/**/test/*.js*
../calypso-prettier/bin/prettier.js  --write client/**/test/*.js*
```

### CODE REVIEW

**Comment by samouri** [comment]
Related to: https://github.com/Automattic/wp-calypso/issues/12260

**Comment by samouri** [comment]
Problem: as of now, prettier isn't merging the docblocks for jest and prettier.

The fix isn't too bad, I'm working on it now

File: client/blocks/daily-post-button/test/helper.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 /**
  * External Dependencies
+ *
+ * @format
```
Comment (gziolo) [inline client/blocks/daily-post-button/test/helper.js]: Whoops :)

**Comment by gziolo** [comment]
I see also 3 failing unit tests:

```
FAIL  client/state/users/suggestions/test/actions.js
  ● Test suite failed to run

    SyntaxError: Unexpected token / in JSON at position 3
        at Object.parse (native)
      
      at Runtime.requireModule (node_modules/jest-runtime/build/index.js:325:61)
      at Object.<anonymous> (client/state/users/suggestions/test/actions.js:24:163)
      at next (native)

FAIL  client/state/reader/site-blocks/test/actions.js
  ● Test suite failed to run

    SyntaxError: Unexpected token / in JSON at position 3
        at Object.parse (native)
      
      at Runtime.requireModule (node_modules/jest-runtime/build/index.js:325:61)
      at Object.<anonymous> (client/state/reader/site-blocks/test/actions.js:24:178)
      at next (native)

 FAIL  client/state/reader/tags/images/test/actions.js
  ● Test suite failed to run

    SyntaxError: Unexpected token / in JSON at position 3
        at Object.parse (native)
      
      at Runtime.requireModule (node_modules/jest-runtime/build/index.js:325:61)
      at Object.<anonymous> (client/state/reader/tags/images/test/actions.js:20:178)
      at next (native)
```

**Comment by gziolo** [comment]
It updated also `.json` files which are located in the test folder ... Example: `client/state/users/suggestions/test/sample-response.json`. I think it would make sense to move them to data subfolder. There are only 3 such files.

**Comment by blowery** [comment]
Might run the prettification as 

```
../calypso-prettier/bin/prettier.js  --write server/**/test/*.js
../calypso-prettier/bin/prettier.js  --write server/**/test/*.jsx
../calypso-prettier/bin/prettier.js  --write client/**/test/*.js
../calypso-prettier/bin/prettier.js  --write client/**/test/*.jsx
```

which would avoid `*.json` files

File: client/blocks/daily-post-button/test/helper.js (hunk 1 above)
Comment (samouri) [reply client/blocks/daily-post-button/test/helper.js]: Ugh.  This one isn't fun to fix.  Looking into if i can have `sort-imports` so the trick: https://github.com/Automattic/wp-calypso/pull/18486

**Comment by samouri** [comment]
@blowery: I think I can use `?` as a single char wildcard instead of `*`.  So that would bring the lines back to:
```bash
../calypso-prettier/bin/prettier.js  --write server/**/test/*.js?
../calypso-prettier/bin/prettier.js  --write client/**/test/*.js?
```

**Comment by dmsnell** [comment]
> It updated also .json files which are located in the test folder ... Example: client/state/users/suggestions/test/sample-response.json. I think it would make sense to move them to data subfolder. There are only 3 such files.

Why is this a problem @gziolo? Isn't it supposed to be a safe way to format JSON as well?

**Comment by samouri** [comment]
The new way I generated this was by:

1. format everything with prettier.  this sometimes combined the `@format` pragma with the `External dependencies` block.  While its technically still valid, it feels a bit strange. Because if we ever remove all external deps and wanted to remove the block we would want to make sure that we kept the `@format`.  I'm not convinced this is actually a problem.
2. run a modified [sort-imports](https://github.com/Automattic/wp-calypso/pull/18486)
3. rerun prettier

**Comment by sirreal** [comment]
Couldn't resist, if you're looking for a one-liner for prettier you can do something like the following, which parallelizes across 4 processes `-P 4`:

```bash
echo {client,server}/**/test/*.{js,jsx} | xargs -n 500 -P 4 ./node_modules/.bin/prettier --write
```

Seeing 931 changes on `master` at 3945ee9b3f2c8bc6a682ea83ccbfc26d84fd287c

**Comment by gziolo** [comment]
@sirreal do you want to rebase this one? I'm happy to review and merge if everything goes well :)

**Comment by sirreal** [comment]
> @sirreal do you want to rebase this one? I'm happy to review and merge if everything goes well :)

@gziolo I'm attempting…

**Comment by samouri** [comment]
All tests pass but ESLint is failing with these errors:
```
```
/home/ubuntu/wp-calypso/client/state/posts/counts/test/selectors.js
  318:12  error  Unquoted reserved word 'private' used as key  quote-props

/home/ubuntu/wp-calypso/client/my-sites/domains/map-domain/test/map-domain.js
  25:8  error  'pageSpy' is already declared in the upper scope  no-shadow

/home/ubuntu/wp-calypso/client/extensions/woocommerce/state/sites/settings/general/test/handlers.js
  29:3  error  Unquoted reserved word 'default' used as key  quote-props

/home/ubuntu/wp-calypso/client/lib/highlight/test/index.js
  14:1  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/components/root-child/test/index.jsx
  61:4  error  Unexpected var, use let or const instead  no-var
/home/ubuntu/wp-calypso/client/state/themes/test/selectors.js
  1013:6  error  Test title is used multiple times in the same test suite  jest/no-identical-title

/home/ubuntu/wp-calypso/client/lib/media/test/utils.js
   38:2   error  Unquoted reserved word 'transient' used as key  quote-props
   46:2   error  Unquoted reserved word 'transient' used as key  quote-props
  128:4   error  Unexpected var, use let or const instead        no-var
  260:4   error  Unexpected var, use let or const instead        no-var
  266:4   error  Unexpected var, use let or const instead        no-var
  362:4   error  Unexpected var, use let or const instead        no-var
  662:19  error  Unquoted reserved word 'transient' used as key  quote-props

/home/ubuntu/wp-calypso/client/state/themes/test/utils.js
  475:5  error  Test title is used multiple times in the same test suite  jest/no-identical-title

/home/ubuntu/wp-calypso/client/lib/plugins/test/utils.js
   96:1  error  Line 96 exceeds the maximum line length of 140   max-len
  106:1  error  Line 106 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/components/tinymce/plugins/media/restrict-size/test/index.js
   55:1  error  Line 55 exceeds the maximum line length of 140   max-len
   58:1  error  Line 58 exceeds the maximum line length of 140   max-len
   64:1  error  Line 64 exceeds the maximum line length of 140   max-len
   73:1  error  Line 73 exceeds the maximum line length of 140   max-len
   84:1  error  Line 84 exceeds the maximum line length of 140   max-len
   87:1  error  Line 87 exceeds the maximum line length of 140   max-len
   93:1  error  Line 93 exceeds the maximum line length of 140   max-len
   96:1  error  Line 96 exceeds the maximum line length of 140   max-len
  102:1  error  Line 102 exceeds the maximum line length of 140  max-len
  105:1  error  Line 105 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/lib/form-state/test/index.js
   77:5  error  Unexpected var, use let or const instead  no-var
   96:5  error  Unexpected var, use let or const instead  no-var
  121:6  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/extensions/woocommerce/state/sites/settings/general/test/selectors.js
  48:2  error  Unquoted reserved word 'default' used as key  quote-props

/home/ubuntu/wp-calypso/client/lib/plugins/wporg-data/test/actions.js
  40:1  error  Line 40 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/my-sites/media-library/test/index.jsx
  21:66  error  Multiple spaces found before ')'  no-multi-spaces
  22:51  error  Multiple spaces found before ')'  no-multi-spaces

/home/ubuntu/wp-calypso/client/state/selectors/test/is-transient-media.js
  56:41  error  Unquoted reserved word 'transient' used as key  quote-props

/home/ubuntu/wp-calypso/client/lib/help-search/test/store.js
  7:1  error  Unexpected var, use let or const instead  no-var
  
  
/home/ubuntu/wp-calypso/client/components/token-field/test/index.jsx
  34:2  error  Unquoted reserved word 'delete' used as key  quote-props

/home/ubuntu/wp-calypso/client/components/infinite-list/test/scroll-helper.js
  412:4  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/lib/media/test/actions.js
  245:7  error  Unquoted reserved word 'transient' used as key  quote-props

/home/ubuntu/wp-calypso/client/post-editor/media-modal/test/index.jsx
   28:2   error  Unquoted reserved word 'default' used as key    quote-props
   48:2   error  Unquoted reserved word 'delete' used as key     quote-props
  364:59  error  Unquoted reserved word 'transient' used as key  quote-props
  365:59  error  Unquoted reserved word 'transient' used as key  quote-props
  391:65  error  Unquoted reserved word 'transient' used as key  quote-props
  416:65  error  Unquoted reserved word 'transient' used as key  quote-props

/home/ubuntu/wp-calypso/client/post-editor/media-modal/test/markup.js
   81:1  error  Line 81 exceeds the maximum line length of 140   max-len
   98:4  error  Unexpected var, use let or const instead         no-var
  100:1  error  Line 100 exceeds the maximum line length of 140  max-len
  105:1  error  Line 105 exceeds the maximum line length of 140  max-len
  134:1  error  Line 134 exceeds the maximum line length of 140  max-len
  189:5  error  Unexpected var, use let or const instead         no-var
  203:1  error  Line 203 exceeds the maximum line length of 140  max-len
  216:1  error  Line 216 exceeds the maximum line length of 140  max-len
  242:1  error  Line 242 exceeds the maximum line length of 140  max-len
  265:1  error  Line 265 exceeds the maximum line length of 140  max-len
  280:1  error  Line 280 exceeds the maximum line length of 140  max-len
  285:5  error  Unexpected var, use let or const instead         no-var
  300:1  error  Line 300 exceeds the maximum line length of 140  max-len
  312:6  error  Unquoted reserved word 'transient' used as key   quote-props
  316:1  error  Line 316 exceeds the maximum line length of 140  max-len
  328:6  error  Unquoted reserved word 'transient' used as key   quote-props
  332:1  error  Line 332 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/lib/media-serialization/test/index.js
  21:1  error  Line 21 exceeds the maximum line length of 140  max-len
  40:1  error  Line 40 exceeds the maximum line length of 140  max-len
  80:1  error  Line 80 exceeds the maximum line length of 140  max-len
  88:1  error  Line 88 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/extensions/woocommerce/woocommerce-services/state/packages/test/actions.js
  64:4  error  Unquoted reserved word 'package' used as key  quote-props

/home/ubuntu/wp-calypso/client/components/tinymce/plugins/contact-form/test/shortcode.js
  111:1  error  Line 111 exceeds the maximum line length of 140  max-len
  167:1  error  Line 167 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/post-editor/editor-discussion/test/index.jsx
  55:4  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/post-editor/editor-ground-control/test/index.jsx
   81:4  error  Unexpected var, use let or const instead  no-var
   89:4  error  Unexpected var, use let or const instead  no-var
  103:4  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/post-editor/editor-categories-tags/test/accordion.jsx
   60:38  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
   68:38  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
   77:31  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
   85:38  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
  100:12  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
  122:12  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
  136:38  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
  156:31  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
  168:38  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace
  180:38  error  className should follow CSS namespace guidelines (expected test__ prefix)  wpcalypso/jsx-classname-namespace

/home/ubuntu/wp-calypso/client/extensions/woocommerce/state/sites/settings/general/test/reducer.js
  45:6  error  Unquoted reserved word 'default' used as key  quote-props

/home/ubuntu/wp-calypso/client/lib/feed-post-store/test/index.js
  15:1  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/lib/mixins/searchable/test/index.js
  22:6  error  Unquoted reserved word 'private' used as key  quote-props
  34:6  error  Unquoted reserved word 'private' used as key  quote-props

/home/ubuntu/wp-calypso/client/my-sites/checkout/cart/test/cart-buttons.js
  22:8  error  'recordStub' is already declared in the upper scope  no-shadow

/home/ubuntu/wp-calypso/client/lib/user/test/utils.js
  21:8  error  'configMock' is already declared in the upper scope  no-shadow

/home/ubuntu/wp-calypso/client/state/selectors/test/get-selected-or-primary-site-id.js
  41:3  error  Test title is used multiple times in the same test suite  jest/no-identical-title

/home/ubuntu/wp-calypso/client/lib/post-normalizer/test/index.js
   24:31  error  There must be a space inside this paren          space-in-parens
   24:52  error  There must be a space inside this paren          space-in-parens
  435:1   error  Line 435 exceeds the maximum line length of 140  max-len
  452:1   error  Line 452 exceeds the maximum line length of 140  max-len
  469:1   error  Line 469 exceeds the maximum line length of 140  max-len
  505:1   error  Line 505 exceeds the maximum line length of 140  max-len
  539:1   error  Line 539 exceeds the maximum line length of 140  max-len

/home/ubuntu/wp-calypso/client/state/themes/test/actions.js
  582:5  error  Test title is used multiple times in the same test suite  jest/no-identical-title

/home/ubuntu/wp-calypso/client/extensions/woocommerce/state/sites/settings/products/test/selectors.js
  49:2  error  Unquoted reserved word 'default' used as key  quote-props
  56:2  error  Unquoted reserved word 'default' used as key  quote-props

/home/ubuntu/wp-calypso/client/components/forms/multi-checkbox/test/index.jsx
  20:2  error  Unexpected var, use let or const instead  no-var
  28:4  error  Unexpected var, use let or const instead  no-var
  44:4  error  Unexpected var, use let or const instead  no-var
  58:4  error  Unexpected var, use let or const instead  no-var
  72:4  error  Unexpected var, use let or const instead  no-var
  91:4  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/blocks/post-status/test/index.jsx
  46:5  error  Test title is used multiple times in the same test suite  jest/no-identical-title
  63:5  error  Test title is used multiple times in the same test suite  jest/no-identical-title
  80:5  error  Test title is used multiple times in the same test suite  jest/no-identical-title
  97:5  error  Test title is used multiple times in the same test suite  jest/no-identical-title

/home/ubuntu/wp-calypso/client/lib/oauth-store/test/index.js
  14:1  error  Unexpected var, use let or const instead  no-var

/home/ubuntu/wp-calypso/client/extensions/woocommerce/state/sites/settings/products/test/actions.js
  36:8  error  Unquoted reserved word 'default' used as key  quote-props
  70:8  error  Unquoted reserved word 'default' used as key  quote-props

/home/ubuntu/wp-calypso/client/components/forms/range/test/index.jsx
  26:3  error  Unexpected var, use let or const instead  no-var
  33:3  error  Unexpected var, use let or const instead  no-var
  43:3  error  Unexpected var, use let or const instead  no-var
  50:3  error  Unexpected var, use let or const instead  no-var
  60:3  error  Unexpected var, use let or const instead  no-var


```

**Comment by gziolo** [comment]
I fixed one conflicting file that got updated. Everything else looks good to go. Thanks for listing all eslint issues. They existed before so we can ignore them for now.

**Pull request merged**